jmclient/jmclient/wallet_rpc.py

```diff
@@ -0,0 +1,659 @@
+
+    def stopService(self):
+        """ Encapsulates shut down actions.
+        """
+        # Currently valid authorization tokens must be removed
+        # from the daemon:
+        self.cookie = None
+        # if the wallet-daemon is shut down, all services
+        # it encapsulates must also be shut down.
+        for name, service in self.services.items():
+            if service:
+                service.stopService()
+        super().stopService()
+
+    @app.handle_errors(NotAuthorized)
+    def not_authorized(self, request, failure):
+        request.setResponseCode(401)
+        return "Invalid credentials."
+
+    @app.handle_errors(NoWalletFound)
+    def no_wallet_found(self, request, failure):
+        request.setResponseCode(404)
+        return "No wallet loaded."
+
+    @app.handle_errors(BackendNotReady)
+    def backend_not_ready(self, request, failure):
+        request.setResponseCode(503)
+        return "Backend daemon not available"
+
+    @app.handle_errors(InvalidRequestFormat)
+    def invalid_request_format(self, request, failure):
+        request.setResponseCode(400)
+        return "Invalid request format."
+
+    @app.handle_errors(ServiceAlreadyStarted)
+    def service_already_started(self, request, failure):
+        request.setResponseCode(401)
+        return "Service already started."
+
+    @app.handle_errors(WalletAlreadyUnlocked)
+    def wallet_already_unlocked(self, request, failure):
+        request.setResponseCode(401)
+        return "Wallet already unlocked."
+
+    @app.handle_errors(ServiceNotStarted)
+    def service_not_started(self, request, failure):
+        request.setResponseCode(401)
+        return "Service cannot be stopped as it is not running."
+
+    def check_cookie(self, request):
+        #part after bearer is what we need
+        auth_header=((request.getHeader('Authorization')))
+        request_cookie = None
+        if auth_header is not None:
+            request_cookie=auth_header[7:]
+        if request_cookie==None or self.cookie != request_cookie:
+            jlog.warn("Invalid cookie: " + str(
+                request_cookie) + ", request rejected.")
+            raise NotAuthorized()
+
+    def get_POST_body(self, request, keys):
+        """ given a request object, retrieve values corresponding
+        to keys keys in a dict, assuming they were encoded using JSON.
+        If *any* of the keys are not present, return False, else
+        returns a dict of those key-value pairs.
+        """
+        assert isinstance(request.content, BytesIO)
+        # we swallow any formatting failure here:
+        try:
+            json_data = json.loads(request.content.read().decode(
+                "utf-8"))
+            retval = {}
+            for k in keys:
+                if k in json_data:
+                    retval[k] = json_data[k]
+                else:
+                    return False
+        except:
+            return False
+        return retval
```

Comment:
```suggestion
            return {k: json_data[k] for k in keys}
        except:
            return False
```
What about this? A missing key raises `key error`, which is caught by the except block and returns False.

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/996/commits/938aa7659ff44d7621294e44bb8a244ccf1b00dd